lib/contracts/contracts.js

```diff
@@ -58,7 +59,20 @@ class ContractsManager {
 
   build(done) {
     let self = this;
+    self.contracts = {};
     async.waterfall([
+      function loadContractFiles(callback) {
+        self.events.request("config:contractsFiles", (contractsFiles) => {
+          self.contractsFiles = contractsFiles;
+          callback();
+        });
+      },
+      function loadContractConfigs(callback) {
+        self.events.request("config:contractsConfig", (contractsConfig) => {
+          self.contractsConfig = cloneDeep(contractsConfig);
+          callback();
+        });
+      },
```

Comment:
I think you can remove the event listener in the constructor then if we reset those here. (check lines 24-29)